### PR TITLE
drop old ztest API

### PR DESCRIPTION
**Pull request by nashif**
- ztest: enable ZTEST_NEW_API by default
- tests: remove CONFIG_ZTEST_NEW_API in all tests
- cmsis_dsp: remove handling of old ztest API
- tests: sensor: do not depend on ZTEST_NEW_API
- drivers: fake: remove special handling for new ztest APIs
- ztest: remove old ztest api
- doc:ztest:  adapt ztest docs

### CODE REVIEW

File: subsys/logging/Kconfig.processing [hunk 1]
```diff
@@ -7,7 +7,7 @@ if !LOG_MODE_MINIMAL
 
 config LOG_PRINTK
 	bool "Process printk messages"
-	default y if PRINTK && (!ZTEST || ZTEST_NEW_API)
+	default y if PRINTK && ZTEST
```
Comment (nordicjm) [inline subsys/logging/Kconfig.processing:10]: This is wrong, should just drop `&& ZTEST`

**Review by teburd — DISMISSED** [review]

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by ghost — COMMENTED** [review]
tests: net: l2: ieee802154: LGTM

**Review by finikorg — DISMISSED** [review]

File: drivers/can/can_fake.c [hunk 2]
```diff
@@ -46,7 +43,6 @@ DEFINE_FAKE_VOID_FUNC(fake_can_set_state_change_callback, const struct device *,
 
 DEFINE_FAKE_VALUE_FUNC(int, fake_can_get_max_filters, const struct device *, bool);
 
-#ifdef CONFIG_ZTEST_NEW_API
```
Comment (henrikbrixandersen) [inline drivers/can/can_fake.c:49]: The "fake" drivers stiil need some kind of `#ifdef` as they can be used outside the Ztest framework as well.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Review by nordicjm — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
Looks like our bot missed the manifest change in this PR. Marking as DNM until https://github.com/zephyrproject-rtos/zscilib/pull/55 is merged and the manifest updated.

**Comment by nashif** [comment]
> Looks like our bot missed the manifest change in this PR. Marking as DNM until [zephyrproject-rtos/zscilib#55](https://github.com/zephyrproject-rtos/zscilib/pull/55) is merged and the manifest updated.

hmm, interesting, @carlescufi  any idea why it was missed?

**Comment by carlescufi** [comment]
> > Looks like our bot missed the manifest change in this PR. Marking as DNM until [zephyrproject-rtos/zscilib#55](https://github.com/zephyrproject-rtos/zscilib/pull/55) is merged and the manifest updated.
> 
> hmm, interesting, @carlescufi any idea why it was missed?

Just identified the issue, will send a PR

**Comment by microbuilder** [comment]
> > Looks like our bot missed the manifest change in this PR. Marking as DNM until [zephyrproject-rtos/zscilib#55](https://github.com/zephyrproject-rtos/zscilib/pull/55) is merged and the manifest updated.
> 
> hmm, interesting, @carlescufi any idea why it was missed?

Merged

**Review by henrikbrixandersen — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]
Flash, VFS and MCUmgr OK.

**Review by glarsennordic — APPROVED** [review]
conn_mgr changes LGTM. No comment on other changes

**Review by cfriedt — APPROVED** [review]

**Pull request merged**